### PR TITLE
Update express-graphql to version 0.5.3

### PR DESCRIPTION
**Pull request by perlerd**
The update of express-graphql to version 0.5.3 caused [this issue](https://github.com/graphql/express-graphql/issues/96). This change should fix this issue. To fix it, I followed [the suggestions](https://github.com/Microsoft/TypeScript/issues/5565#issuecomment-155216760) made by [Anders Hejlsberg](https://github.com/ahejlsberg).

### CODE REVIEW

File: express-graphql/express-graphql.d.ts [hunk 1]
```diff
@@ -8,45 +8,48 @@
...
+        };
+
+        type Middleware = (request:Request, response:Response) => void;
+    }
 
     /**
      * Middleware for express; takes an options object or function as input to
      * configure behavior, and returns an express middleware.
      */
-    export default function graphqlHTTP(options: Options): Middleware;
```
Comment (mhegazy) [inline express-graphql/express-graphql.d.ts]: the library seems to have a default export in https://github.com/graphql/express-graphql/blob/master/src/index.js#L85. why is this removed?

Comment (perlerd) [inline express-graphql/express-graphql.d.ts]: I replaced it to enable the use of ES6 syntax for imports. With the current implementation, using ES6 syntax for importing (for example `import * as graphqlHTTP from "express-graphql";`) fails with compile error `TS2349: Cannot invoke an expression whose type lacks a call signature.`. The proposed implementation will enable the use ES6 syntax to import this module in Typescript. The new implementation follows the suggestion made in [this discussion](https://github.com/Microsoft/TypeScript/issues/5565).

Suggestion:
Replace the import statement in the test file _express-graphql-tests.ts_ by
```
import * as graphqlHTTP from "express-graphql";
```

**Comment by isman-usoh** [comment]
@perlerd i update express-graphql

express-graphql/express-graphql.d.ts 
```
// Type definitions for express-graphql
// Project: https://www.npmjs.org/package/express-graphql
// Definitions by: Isman Usoh <https://github.com/isman-usoh>, Nitin Tutlani <https://github.com/nitintutlani>
// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped

/// <reference path="../express/express.d.ts" />

declare namespace ExpressGraphQL {
    import E = Express;

    type Middleware = (request: E.Request, response: E.Response) => void;

    /**
     * Used to configure the graphQLHTTP middleware by providing a schema
     * and other configuration options.
     */
    type Options = ((req: E.Request) => OptionsObj) | OptionsObj
    type OptionsObj = {
        /**
         * A GraphQL schema from graphql-js.
         */
        schema: Object,

        /**
         * A value to pass as the context to the graphql() function.
         */
        context?: any,

        /**
         * An object to pass as the rootValue to the graphql() function.
         */
        rootValue?: Object,

        /**
         * A boolean to configure whether the output should be pretty-printed.
         */
        pretty?: boolean,

        /**
         * An optional function which will be used to format any errors produced by
         * fulfilling a GraphQL operation. If no function is provided, GraphQL's
         * default spec-compliant `formatError` function will be used.
         */
        formatError?: Function,

        /**
         * An optional array of validation rules that will be applied on the document
         * in additional to those defined by the GraphQL spec.
         */
        validationRules?: Array<any>,

        /**
         * A boolean to optionally enable GraphiQL mode.
         */
        graphiql?: boolean,
    };
}

declare module "express-graphql" {
    const graphqlHTTP: {
        (options: ExpressGraphQL.Options): ExpressGraphQL.Middleware
    }
    export = graphqlHTTP;
}
```

express-graphql/express-graphql-tests.ts
```
/// <reference path="./express-graphql.d.ts" />
/// <reference path="../express/express.d.ts" />
/// <reference path="../express-session/express-session.d.ts" />

import * as express from "express";
import * as graphqlHTTP from "express-graphql";

const app = express();
const schema = {};

const graphqlOption: ExpressGraphQL.OptionsObj = {
    graphiql: true,
    schema: schema,
    formatError: (error) => ({
        message: error.message,
        locations: error.locations,
        stack: error.stack
    })
}

const graphqlOptionRequest = (request: express.Request): ExpressGraphQL.OptionsObj => ({
    graphiql: true,
    schema: schema,
    context: request.session,
});

app.use("/graphql1", graphqlHTTP(graphqlOption));

app.use("/graphql2", graphqlHTTP(graphqlOptionRequest));

app.listen(8080);
```

File: express-graphql/express-graphql.d.ts (hunk 1 above)
Comment (mhegazy) [inline express-graphql/express-graphql.d.ts]: i see. thanks for the explanation.

**Comment by mhegazy** [comment]
Any comments @isman-usoh and @nitintutlani?

**Comment by perlerd** [comment]
@isman-usoh I agree to your solution. Thanks for updating.

**Comment by isman-usoh** [comment]
@perlerd update and commit instead of me, please?

**Comment by perlerd** [comment]
@isman-usoh: With the current implementation, it is possible to access the declared types using `import {Middleware, Options, OptionsObj} from "express-graphql";`. This will not be possible anymore with your proposed solution as you moved these types to a separate namespace. Could you please explain the motivation to put them into a separate namespace?

I have done the following updates:
- changed _express-graphql/express-graphql-tests.ts_ using your suggestion with some minor adaptations
- added the property _context_ to the type _OptionsObj_ in _express-graphql/express-graphql.d.ts_ (which is still based on my proposed solution)

**Comment by isman-usoh** [comment]
namespaces very simple use and namespaces needless to declare import before use
ex. ExpressGraphQL.OptionsObj

http://www.typescriptlang.org/docs/handbook/namespaces-and-modules.html#using-namespaces

**Comment by mhegazy** [comment]
Thanks for the changes, and thanks @isman-usoh for the review.
